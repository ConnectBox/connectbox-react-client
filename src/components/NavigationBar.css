.breadcrumb {
    background-color: #CCEFEB;
}

.navbar-container {
    display: flex;
    flex: 1;
    flex-direction: row;
}

.navbar-element {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@keyframes rotate {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
}

.loader {
    animation: rotate 1s infinite linear;
}
