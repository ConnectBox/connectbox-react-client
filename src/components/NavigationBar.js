import './NavigationBar.css'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

class NavigationBar extends Component {
    static propTypes = {
        contentPath: PropTypes.string.isRequired,
        loading: PropTypes.bool.isRequired,
        reload: PropTypes.func.isRequired
    }

    constructor (props) {
        super(props)

        this.pathParts = this.pathParts.bind(this)
    }

    pathParts () {
      const { contentPath } = this.props
      let parts = (contentPath === '/' ? '': contentPath).split('/')
      let up = ''
      for (var i = 0, l = parts.length; i < l; i++) {
        const path = up + parts[i]
        up = path + '/'
        parts[i] = { path: path, name: parts[i] || '<root>' }
      }
      return parts
    }

  render () {
    let pathElements = []
    const pathParts = this.pathParts()
    for (let i = 0; i < pathParts.length; i++) {
      const part = pathParts[i]
      const className = part.name !== '<root>' ? 'navbar-element' : null;
      const style = {};
      if (part.name !== '<root>') {
        if (i === pathParts.length - 1) {
          style['flex-basis'] = '100%';
        }
      }
      pathElements.push(
        <li key={`crumb${i}`} className={className} style={style}>
          {part.name === '<root>' && 
          <span>
           <Link to=''><i className="fa fa-home fa-lg"></i></Link>
          </span>}

          {i === pathParts.length - 1 && part.name !== '<root>' &&
          <span>{part.name}</span>}

          {i < pathParts.length - 1 && part.name !== '<root>' &&
          <Link to={`/#${part.path}`}>{part.name}</Link>}
        </li>
      )
    }

    return (
      <ol className='navbar-container breadcrumb' style={{left: '0px', right: '0px', position: 'absolute', zIndex: 1000}}>
        <li>
          <a role='button' onClick={this.props.reload} title='Reload'>
            <i className={`fa fa-refresh${this.props.loading ? ' loader' : ''}`} />
          </a>
        </li>
        {pathElements}
      </ol>
    )
  }
}

export default NavigationBar
