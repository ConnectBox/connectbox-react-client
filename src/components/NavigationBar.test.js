import React from 'react'
import NavigationBar from './NavigationBar'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'

test('Render NavigationBar Root', () => {
  const component = renderer.create(
    <Router>
      <NavigationBar
        contentPath='/content'
        reload={() => {}}
        loading={false} />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render NavigationBar Subfolder', () => {
  const component = renderer.create(
    <Router>
      <NavigationBar
        contentPath='/content/foo/bar'
        reload={() => {}}
        loading={false} />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render NavigationBar Root - Loading', () => {
  const component = renderer.create(
    <Router>
      <NavigationBar
        contentPath='/content'
        reload={() => {}}
        loading />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
