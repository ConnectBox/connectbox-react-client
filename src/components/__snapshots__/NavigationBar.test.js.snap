// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavigationBar Root - Loading 1`] = `
<ol
  className="breadcrumb"
  style={
    Object {
      "left": "20px",
      "position": "fixed",
      "right": "20px",
      "zIndex": 1000,
    }
  }
>
  <li>
    <a
      onClick={[Function]}
      role="button"
      title="Reload"
    >
      <i
        className="fa fa-refresh loader"
      />
    </a>
  </li>
  <li>
    <span>
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-home fa-lg"
        />
      </a>
    </span>
  </li>
  <li>
    <span>
      content
    </span>
  </li>
</ol>
`;

exports[`Render NavigationBar Root 1`] = `
<ol
  className="breadcrumb"
  style={
    Object {
      "left": "20px",
      "position": "fixed",
      "right": "20px",
      "zIndex": 1000,
    }
  }
>
  <li>
    <a
      onClick={[Function]}
      role="button"
      title="Reload"
    >
      <i
        className="fa fa-refresh"
      />
    </a>
  </li>
  <li>
    <span>
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-home fa-lg"
        />
      </a>
    </span>
  </li>
  <li>
    <span>
      content
    </span>
  </li>
</ol>
`;

exports[`Render NavigationBar Subfolder 1`] = `
<ol
  className="breadcrumb"
  style={
    Object {
      "left": "20px",
      "position": "fixed",
      "right": "20px",
      "zIndex": 1000,
    }
  }
>
  <li>
    <a
      onClick={[Function]}
      role="button"
      title="Reload"
    >
      <i
        className="fa fa-refresh"
      />
    </a>
  </li>
  <li>
    <span>
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-home fa-lg"
        />
      </a>
    </span>
  </li>
  <li>
    <a
      href="/#/content"
      onClick={[Function]}
    >
      content
    </a>
  </li>
  <li>
    <a
      href="/#/content/foo"
      onClick={[Function]}
    >
      foo
    </a>
  </li>
  <li>
    <span>
      bar
    </span>
  </li>
</ol>
`;
